src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -753,7 +753,7 @@ private List<StudentParticipation> filterParticipationsWithRelevantResults(List<
                 .filter(participation -> participation.getParticipant() != null)
 
                 // filter all irrelevant results, i.e. rated = false or no completion date or no score
-                .peek(participation -> {
+                .map(participation -> {
```

Comment:
Why this change?
Maybe we could change this whole block to directly calling removeIf on the list and then simply iterating over the entries that are left in the list